pkg/vz/vm_darwin.go

```diff
@@ -0,0 +1,485 @@
+		return nil, err
+	}
+
+	if err = attachDisplay(driver, vmConfig); err != nil {
+		return nil, err
+	}
+
+	if err = attachConsole(driver, vmConfig); err != nil {
+		return nil, err
+	}
+
+	if err = attachFolderMounts(driver, vmConfig); err != nil {
+		return nil, err
+	}
+
+	if err = attachOtherDevices(driver, vmConfig); err != nil {
+		return nil, err
+	}
+
+	validated, err := vmConfig.Validate()
+	if !validated || err != nil {
+		return nil, err
+	}
+
+	return vz.NewVirtualMachine(vmConfig)
+}
+
+func createInitialConfig(driver *driver.BaseDriver) (*vz.VirtualMachineConfiguration, error) {
+	efiVariableStore, err := getEFI(driver)
+	if err != nil {
+		return nil, err
+	}
+
+	bootLoader, err := vz.NewEFIBootLoader(vz.WithEFIVariableStore(efiVariableStore))
+	if err != nil {
+		return nil, err
+	}
+
+	bytes, err := units.RAMInBytes(*driver.Yaml.Memory)
+	if err != nil {
+		return nil, err
+	}
+
+	vmConfig, err := vz.NewVirtualMachineConfiguration(
+		bootLoader,
+		uint(*driver.Yaml.CPUs),
+		uint64(bytes),
+	)
+	if err != nil {
+		return nil, err
+	}
+	return vmConfig, nil
+}
+
+func attachPlatformConfig(driver *driver.BaseDriver, vmConfig *vz.VirtualMachineConfiguration) error {
+	machineIdentifier, err := getMachineIdentifier(driver)
+	if err != nil {
+		return err
+	}
+
+	platformConfig, err := vz.NewGenericPlatformConfiguration(vz.WithGenericMachineIdentifier(machineIdentifier))
+	if err != nil {
+		return err
+	}
+	vmConfig.SetPlatformVirtualMachineConfiguration(platformConfig)
+	return nil
+}
+
+func attachSerialPort(driver *driver.BaseDriver, config *vz.VirtualMachineConfiguration) error {
+	path := filepath.Join(driver.Instance.Dir, filenames.SerialLog)
+	serialPortAttachment, err := vz.NewFileSerialPortAttachment(path, false)
+	if err != nil {
+		return err
+	}
+	consoleConfig, err := vz.NewVirtioConsoleDeviceSerialPortConfiguration(serialPortAttachment)
+	config.SetSerialPortsVirtualMachineConfiguration([]*vz.VirtioConsoleDeviceSerialPortConfiguration{
+		consoleConfig,
+	})
+	return err
+}
+
+func attachNetwork(driver *driver.BaseDriver, vmConfig *vz.VirtualMachineConfiguration, networkConn *os.File) error {
+	//slirp network using gvisor netstack
+	fileAttachment, err := vz.NewFileHandleNetworkDeviceAttachment(networkConn)
+	if err != nil {
+		return err
+	}
+	err = fileAttachment.SetMaximumTransmissionUnit(1500)
+	if err != nil {
+		return err
+	}
+	networkConfig, err := vz.NewVirtioNetworkDeviceConfiguration(fileAttachment)
+	if err != nil {
+		return err
+	}
+	mac, err := net.ParseMAC(limayaml.MACAddress(driver.Instance.Dir))
+	if err != nil {
+		return err
+	}
+	address, err := vz.NewMACAddress(mac)
+	if err != nil {
+		return err
+	}
+	networkConfig.SetMACAddress(address)
+
+	configurations := []*vz.VirtioNetworkDeviceConfiguration{
+		networkConfig,
+	}
+	//Other networks
+	for _, network := range driver.Yaml.Networks {
+		//TODO - Handle NAT network type
+		if network.Socket != "" {
```

Comment:
The `network.Socket` value does not seem consumed

Reply:
I expected it could be supported via VZFileHandleNetworkDeviceAttachment , but seems incompatible with QEMU:
https://developer.apple.com/documentation/virtualization/vzfilehandlenetworkdeviceattachment?language=objc

So we have to error out when `network.Socket` is non-empty

Reply:
I think we can support it, maybe some extra support might be needed from socker_vmnet as well. I will try to support this in a different PR.